server/src/configs/validators.rs

```diff
@@ -20,6 +21,16 @@ impl Validatable<ServerError> for ServerConfig {
         self.system.compression.validate()?;
         self.personal_access_token.validate()?;
 
+        let topic_size = match self.system.retention_policy.max_topic_size {
+            MaxTopicSize::Custom(size) => size.as_bytes_u64(),
+            MaxTopicSize::ServerDefault => MaxTopicSize::get_server_default().as_bytes_u64(),
+        };
```

Comment:
IMHO you should change logic here. return error in case if MaxTopicSize is set to server default - user should always set it properly, since this is server config validation.

Reply:
Fixed